src/screens/hardware-wallets/PairHardwareWalletSearchSheet.tsx

```diff
@@ -1,27 +1,62 @@
 import * as i18n from '@/languages';
 import React from 'react';
-import { Inset, Stack, Text } from '@/design-system';
+import { Box, Inset, Stack, Text } from '@/design-system';
 import { Layout } from '@/screens/hardware-wallets/components/Layout';
-import { CancelButton } from '@/screens/hardware-wallets/components/CancelButton';
 import { TRANSLATIONS } from '@/screens/hardware-wallets/constants';
+import { useSetRecoilState } from 'recoil';
+import { useLedgerImport } from '@/hooks/useLedgerImport';
+import Routes from '@/navigation/routesNames';
+import { useNavigation } from '@/navigation';
+import { LedgerImportDeviceIdAtom } from '@/navigation/PairHardwareWalletNavigator';
+import { ActionButton } from '@/screens/hardware-wallets/components/ActionButton';
+import { CancelButton } from './components/CancelButton';
 
-export const PairHardwareWalletSearchSheet = () => (
-  <Layout>
-    <Inset horizontal="36px">
-      <Stack alignHorizontal="center" space="20px">
-        <Text align="center" color="label" weight="bold" size="26pt">
-          {i18n.t(TRANSLATIONS.looking_for_devices)}
-        </Text>
-        <Text
-          align="center"
-          color="labelTertiary"
-          weight="semibold"
-          size="15pt / 135%"
-        >
-          {i18n.t(TRANSLATIONS.make_sure_bluetooth_enabled)}
-        </Text>
-      </Stack>
-    </Inset>
-    <CancelButton />
-  </Layout>
-);
+export const PairHardwareWalletSearchSheet = () => {
+  const { navigate, goBack } = useNavigation();
+  const setDeviceId = useSetRecoilState(LedgerImportDeviceIdAtom);
+  const [isConnected, setIsConnected] = React.useState(false);
+
+  useLedgerImport({
+    successCallback: deviceId => {
+      setDeviceId(deviceId);
+      setIsConnected(true);
+    },
+  });
+
+  return (
+    <Layout>
+      <Inset horizontal="36px">
+        <Stack alignHorizontal="center" space="20px">
+          <Text align="center" color="label" weight="bold" size="26pt">
+            {isConnected
+              ? i18n.t(TRANSLATIONS.device_connected)
+              : i18n.t(TRANSLATIONS.looking_for_devices)}
+          </Text>
+          <Text
+            align="center"
+            color="labelTertiary"
+            weight="semibold"
+            size="15pt / 135%"
+          >
+            {isConnected
+              ? i18n.t(TRANSLATIONS.almost_done)
+              : i18n.t(TRANSLATIONS.make_sure_bluetooth_enabled)}
+          </Text>
+        </Stack>
+      </Inset>
+      <Box width="full">
+        <Stack space="16px">
+          {isConnected && (
+            <ActionButton
+              label={i18n.t(i18n.l.button.next)}
+              onPress={() =>
+                navigate(Routes.PAIR_HARDWARE_WALLET_SIGNING_SHEET)
+              }
+            />
+          )}
+          <CancelButton onPress={goBack} />
```

Comment:
i think its pretty weird how this button doesn't exit the navigator, seems kinda pointless since the only action to take after "going back" is to go forward lol. i think we should remove

Reply:
when the device is paired, it shows both the next button and cancel button. if we do decide to keep the cancel button, i think we should replace it with the next button when the device is connected.

Reply:
this was a design ask to re add, ill pass this on as i do agree